app/models/alma_renew/item.rb

```diff
@@ -12,12 +14,35 @@ def initialize(item_hash, from_agency_id: LibJobs.config[:ncip_renew_from_agency
       @user_id = item_hash["Primary Identifier"]
       @item_barcode = item_hash["Barcode"]
       @user_group = item_hash["Patron Group"]
+      @errors = ActiveModel::Errors.new(self)
     end
 
     def ncip
       ncip_renew_body(due_date: calculate_due_date).to_xml
     end
 
+    def valid?
+      validate_expiration_date!
+      validate_user_id!
+      errors.empty?
+    end
+
+    def read_attribute_for_validation(attr)
+      send(attr)
+    end
+
+    def to_h
+      { "Barcode" => item_barcode, "Expiry Date" => expiration_date, "Primary Identifier" => user_id, "Patron Group" => user_group }
+    end
+
+    def self.human_attribute_name(attr, _options = {})
+      attr
+    end
+
+    def self.lookup_ancestors
+      [self]
+    end
```

Comment:
Rather than re-implementing these methods, would it make more sense to just bring in the default rails implementations (it looks like that'd be through `extend ActiveModel::Translation`)?

Reply:
Done!